flutter/news/lib/core/news_api_client.dart

```diff
@@ -0,0 +1,12 @@
+import 'package:dio/dio.dart';
+import 'package:news/features/frontpage/domain/entities/base_news_response.dart';
+import 'package:retrofit/retrofit.dart';
+part 'news_api_client.g.dart';
+
+@RestApi(baseUrl: "https://newsapi.org")
+abstract class NewsApiClient {
```

Comment:
Once we have to use this for real we'll be passing the appropriate headers (the `NEWS_API_KEY`)